proto/def.bzl

```diff
@@ -99,21 +99,23 @@ def _go_proto_library_impl(ctx):
         srcs = go_srcs,
     )
     source = go.library_to_source(go, ctx.attr, library, False)
-    if not valid_archive:
-        return [library, source]
-    archive = go.archive(go, source)
-    return [
-        library,
-        source,
-        archive,
-        DefaultInfo(
-            files = depset([archive.data.file]),
-            runfiles = archive.runfiles,
-        ),
-        OutputGroupInfo(
-            compilation_outputs = [archive.data.file],
-        ),
-    ]
+    providers = [library, source]
+    if valid_archive:
+        archive = go.archive(go, source)
+        providers.extend([
+            archive,
+            DefaultInfo(
+                files = depset([archive.data.file]),
+                runfiles = archive.runfiles,
+            ),
+            OutputGroupInfo(
+                compilation_outputs = [archive.data.file],
+            ),
+        ])
+    return struct(
+        providers = providers,
+        aspect_proto_go_api_info = struct(files_to_build = go_srcs),
```

Comment:
This sounds like it should be produced by an aspect, not by `go_proto_library` itself.

Reply:
I think this is what makes me uncomfortable with this PR -- rule sets should not have to provide anything for specific editors. We can expose general information through Go providers, but supporting specific tools like this is too constraining.

Based on the blame information in that link, since this code was exported out of Google's internal monorepo, I'm guessing this aspect was designed to work with Blaze rules. With Blaze, it's realistic for a single team to curate rules, but I don't think it's realistic for Bazel. Bazel rule sets will have different provider APIs, and there needs to be a plan to deal with that.

Reply:
That's a fair point--and I wonder if there's a better way to approach this. Thus far my search for a solution has yielded problems on both sides of the ruleset<->editor partnership:
- Rule sets cannot be expected to individually accommodate all editors
- Editors can't reasonably require that a user's WORKSPACE include every ruleset that the given editor supports (which is currently the case when editor-relevant build artifacts are accessible only via the new provider API, since the editor's introspection aspect would need to unconditionally load relevant providers from all supported rulesets)

Perhaps the ideal solution is one which:
- Is not editor specific (from a ruleset perspective)
- Does not impose dependencies on a project's WORKSPACE (from an editor perspective)

What do you think about exposing the generated .pb.go sources via an appropriately named OutputGroup (perhaps 'generated_srcs' or 'go_generated_srcs')? This would accomplish the above (and without resorting to the legacy provider API).

Reply:
If an output group works for you, I think that's the best option, and it would probably be useful for other applications as well. If you update the PR with that change, I'd be happy to approve and merge. `go_generated_srcs` is a good name.

Reply:
Updated; generated srcs are now exposed via OutputGroup instead of legacy provider.